src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarEdit.tsx

```diff
@@ -85,169 +94,216 @@ const GanttTaskBarEdit = ({
-            workPackages: [],
-            projectId: event.id,
-            projectNumber: event.projectNumber,
-            carNumber: event.carNumber,
-            stage: workPackage.stage,
-            styles: {
-              color: GanttWorkPackageTextColorPipe(workPackage.stage),
-              backgroundColor: GanttWorkPackageStageColorPipe(workPackage.stage, WbsElementStatus.Inactive)
-            }
-          });
+  const createChangeHandler = (change: GanttChange) => {
+    createChange(change);
+  };
 
-          createChange({
-            id,
-            eventId: dup,
-            type: 'create-work-package',
-            name: workPackage.name,
-            stage: workPackage.stage,
-            start: new Date(),
-            end: new Date(Date.now() + 1000 * 60 * 60 * 24 * 7)
-          });
-        }}
-      />
-      <Box
-        sx={{
-          width: '100%',
-          display: 'grid',
-          gap: GANTT_CHART_GAP_SIZE,
-          gridTemplateRows: `repeat(1, minmax(0, 1fr))`,
-          gridTemplateColumns: `repeat(${days.length}, minmax(auto, 1fr))`,
-          position: 'absolute',
-          top: 0,
-          left: 0
-        }}
-      >
-        {/* Drop areas */}
-        {showDropPoints &&
-          days.map((day, index) => (
-            <Box
-              key={index}
-              onDragOver={onDragOver}
-              onDrop={() => onDrop(day)}
-              sx={{
-                borderRadius: '0.25rem',
-                height: '2rem',
-                minWidth: GANTT_CHART_CELL_SIZE,
-                maxWidth: GANTT_CHART_CELL_SIZE,
-                backgroundColor: `color-mix(in srgb, ${theme.palette.background.default}, transparent 75%);`
-              }}
-            />
-          ))}
-      </Box>
-      <Box
-        sx={{
-          display: 'grid',
-          gap: GANTT_CHART_GAP_SIZE,
-          gridTemplateRows: `repeat(1, minmax(0, 1fr))`,
-          gridTemplateColumns: `repeat(${days.length}, minmax(${GANTT_CHART_CELL_SIZE}, 1fr))`,
-          width: '100%'
-        }}
+  return (
+    <>
+      <div
+        style={{ position: 'relative', width: '100%', marginTop: 10 }}
+        onMouseUp={handleMouseUp}
+        onMouseMove={handleMouseMove}
       >
-        <div
-          ref={measureRef}
-          style={{
-            gridColumnStart: getStartCol(event.start),
-            gridColumnEnd: getEndCol(event.end),
-            height: '2rem',
-            width: width === 0 ? `unset` : `${width}px`,
-            border: `1px solid ${isResizing ? theme.palette.text.primary : theme.palette.divider}`,
-            borderRadius: '0.25rem',
-            backgroundColor: event.styles ? event.styles.backgroundColor : theme.palette.background.paper,
-            cursor: isProject ? 'default' : 'move'
+        <AddWorkPackageModal
+          showModal={showAddWorkPackageModal}
+          handleClose={() => setShowAddWorkPackageModal(false)}
+          addWorkPackage={(workPackageInfo) => {
```

Comment:
this method should be named and moved above the component return to clean up code